Match Summary

In [1]:
import pandas as pd
import json

In [59]:
with open('E:\\BSc.IT\\TY\\DS Project\\t20_json_files\\t20_wc_match_results.json') as f:
    data = json.load(f)
#data[0]["matchSummary"]
len(data)

1

In [60]:
df_match = pd.DataFrame(data[0]['matchSummary'])
df_match.head()

,team1,team2,winner,margin,ground,matchDate,scorecard
0,Namibia,Sri Lanka,Namibia,55 runs,Geelong,"Oct 16, 2022",T20I # 1823
1,Netherlands,U.A.E.,Netherlands,3 wickets,Geelong,"Oct 16, 2022",T20I # 1825
2,Scotland,West Indies,Scotland,42 runs,Hobart,"Oct 17, 2022",T20I # 1826
3,Ireland,Zimbabwe,Zimbabwe,31 runs,Hobart,"Oct 17, 2022",T20I # 1828
4,Namibia,Netherlands,Netherlands,5 wickets,Geelong,"Oct 18, 2022",T20I # 1830


In [61]:
df_match.shape

(45, 7)

In [62]:
df_match.rename({'scorecard': 'match_id'}, axis = 1, inplace = True)
df_match.head()

,team1,team2,winner,margin,ground,matchDate,match_id
0,Namibia,Sri Lanka,Namibia,55 runs,Geelong,"Oct 16, 2022",T20I # 1823
1,Netherlands,U.A.E.,Netherlands,3 wickets,Geelong,"Oct 16, 2022",T20I # 1825
2,Scotland,West Indies,Scotland,42 runs,Hobart,"Oct 17, 2022",T20I # 1826
3,Ireland,Zimbabwe,Zimbabwe,31 runs,Hobart,"Oct 17, 2022",T20I # 1828
4,Namibia,Netherlands,Netherlands,5 wickets,Geelong,"Oct 18, 2022",T20I # 1830


Create a match ids dictionary that maps team names to a unique match id. This will be useful later on to link with other tables

In [63]:
match_ids_dict = {}
for index, row in df_match.iterrows():
    key1 = row['team1'] + " Vs " + row['team2']
    key2 = row['team2'] + " Vs " + row['team1']
    match_ids_dict[key1] = row['match_id']
    match_ids_dict[key2] = row['match_id']
len(match_ids_dict)  

90

In [40]:
df_match.dtypes

team1        object
team2        object
winner       object
margin       object
ground       object
matchDate    object
match_id     object
dtype: object

In [64]:
df_match['matchDate'] = pd.to_datetime(df_match['matchDate'])
df_match.head()

,team1,team2,winner,margin,ground,matchDate,match_id
0,Namibia,Sri Lanka,Namibia,55 runs,Geelong,2022-10-16,T20I # 1823
1,Netherlands,U.A.E.,Netherlands,3 wickets,Geelong,2022-10-16,T20I # 1825
2,Scotland,West Indies,Scotland,42 runs,Hobart,2022-10-17,T20I # 1826
3,Ireland,Zimbabwe,Zimbabwe,31 runs,Hobart,2022-10-17,T20I # 1828
4,Namibia,Netherlands,Netherlands,5 wickets,Geelong,2022-10-18,T20I # 1830


In [65]:
specific_date = pd.to_datetime('2022-10-22')
df_match['stage'] = df_match['matchDate'].apply(lambda x: "Qualifier" if x < specific_date else "Super12")
df_match.tail(10)

,team1,team2,winner,margin,ground,matchDate,match_id,stage
35,Pakistan,South Africa,Pakistan,33 runs,Sydney,2022-11-03,T20I # 1861,Super12
36,Ireland,New Zealand,New Zealand,35 runs,Adelaide,2022-11-04,T20I # 1862,Super12
37,Australia,Afghanistan,Australia,4 runs,Adelaide,2022-11-04,T20I # 1864,Super12
38,England,Sri Lanka,England,4 wickets,Sydney,2022-11-05,T20I # 1867,Super12
39,Netherlands,South Africa,Netherlands,13 runs,Adelaide,2022-11-06,T20I # 1871,Super12
40,Bangladesh,Pakistan,Pakistan,5 wickets,Adelaide,2022-11-06,T20I # 1872,Super12
41,India,Zimbabwe,India,71 runs,Melbourne,2022-11-06,T20I # 1873,Super12
42,New Zealand,Pakistan,Pakistan,7 wickets,Sydney,2022-11-09,T20I # 1877,Super12
43,England,India,England,10 wickets,Adelaide,2022-11-10,T20I # 1878,Super12
44,England,Pakistan,England,5 wickets,Melbourne,2022-11-13,T20I # 1879,Super12


In [66]:
df=df_match[df_match["stage"]=='Qualifier'].count()
df

team1        12
team2        12
winner       12
margin       12
ground       12
matchDate    12
match_id     12
stage        12
dtype: int64

In [67]:
df_match.dtypes

team1                object
team2                object
winner               object
margin               object
ground               object
matchDate    datetime64[ns]
match_id             object
stage                object
dtype: object

In [68]:
df_match.to_csv('cricket_csv_files/match_summary.csv',index=False)

Batting Summary

In [69]:
with open('E:\\BSc.IT\\TY\\DS Project\\t20_json_files\\t20_json_files\\t20_wc_batting_summary.json') as f:
    data = json.load(f) 
    #print(len(data))
    #data[0]["battingSummary"]
    all_records = []
    for rec in data:
        all_records.extend(rec["battingSummary"])

df_batting = pd.DataFrame(all_records)
df_batting.head(11)
#len(df_batting)

,match,teamInnings,battingPos,batsmanName,dismissal,runs,balls,4s,6s,SR
0,Namibia Vs Sri Lanka,Namibia,1,Michael van Lingen,c Pramod Madushan b Chameera,3,6,0,0,50.00
1,Namibia Vs Sri Lanka,Namibia,2,Divan la Cock,c Shanaka b Pramod Madushan,9,9,1,0,100.00
2,Namibia Vs Sri Lanka,Namibia,3,Jan Nicol Loftie-Eaton,c †Mendis b Karunaratne,20,12,1,2,166.66
3,Namibia Vs Sri Lanka,Namibia,4,Stephan Baard,c DM de Silva b Pramod Madushan,26,24,2,0,108.33
4,Namibia Vs Sri Lanka,Namibia,5,Gerhard Erasmus(c),c Gunathilaka b PWH de Silva,20,24,0,0,83.33
5,Namibia Vs Sri Lanka,Namibia,6,Jan Frylinck,run out (Gunathilaka/†Mendis),44,28,4,0,157.14
6,Namibia Vs Sri Lanka,Namibia,7,David Wiese,c †Mendis b Theekshana,0,1,0,0,0.00
7,Namibia Vs Sri Lanka,Namibia,8,JJ Smit,,31,16,2,2,193.75
8,Namibia Vs Sri Lanka,Sri Lanka,1,Pathum Nissanka,c Smit b Shikongo,9,10,1,0,90.00
9,Namibia Vs Sri Lanka,Sri Lanka,2,Kusal Mendisâ€,c †Green b Wiese,6,6,0,0,100.00


In [70]:
df_batting["out/not_out"] = df_batting['dismissal'].apply(lambda x: "out" if len(x)>0 else "not_out")
df_batting.head(11)

,match,teamInnings,battingPos,batsmanName,dismissal,runs,balls,4s,6s,SR,out/not_out
0,Namibia Vs Sri Lanka,Namibia,1,Michael van Lingen,c Pramod Madushan b Chameera,3,6,0,0,50.00,out
1,Namibia Vs Sri Lanka,Namibia,2,Divan la Cock,c Shanaka b Pramod Madushan,9,9,1,0,100.00,out
2,Namibia Vs Sri Lanka,Namibia,3,Jan Nicol Loftie-Eaton,c †Mendis b Karunaratne,20,12,1,2,166.66,out
3,Namibia Vs Sri Lanka,Namibia,4,Stephan Baard,c DM de Silva b Pramod Madushan,26,24,2,0,108.33,out
4,Namibia Vs Sri Lanka,Namibia,5,Gerhard Erasmus(c),c Gunathilaka b PWH de Silva,20,24,0,0,83.33,out
5,Namibia Vs Sri Lanka,Namibia,6,Jan Frylinck,run out (Gunathilaka/†Mendis),44,28,4,0,157.14,out
6,Namibia Vs Sri Lanka,Namibia,7,David Wiese,c †Mendis b Theekshana,0,1,0,0,0.00,out
7,Namibia Vs Sri Lanka,Namibia,8,JJ Smit,,31,16,2,2,193.75,not_out
8,Namibia Vs Sri Lanka,Sri Lanka,1,Pathum Nissanka,c Smit b Shikongo,9,10,1,0,90.00,out
9,Namibia Vs Sri Lanka,Sri Lanka,2,Kusal Mendisâ€,c †Green b Wiese,6,6,0,0,100.00,out


In [71]:
df_batting.drop('dismissal',axis=1,inplace=True)

In [72]:
df_batting.head(11)

,match,teamInnings,battingPos,batsmanName,runs,balls,4s,6s,SR,out/not_out
0,Namibia Vs Sri Lanka,Namibia,1,Michael van Lingen,3,6,0,0,50.00,out
1,Namibia Vs Sri Lanka,Namibia,2,Divan la Cock,9,9,1,0,100.00,out
2,Namibia Vs Sri Lanka,Namibia,3,Jan Nicol Loftie-Eaton,20,12,1,2,166.66,out
3,Namibia Vs Sri Lanka,Namibia,4,Stephan Baard,26,24,2,0,108.33,out
4,Namibia Vs Sri Lanka,Namibia,5,Gerhard Erasmus(c),20,24,0,0,83.33,out
5,Namibia Vs Sri Lanka,Namibia,6,Jan Frylinck,44,28,4,0,157.14,out
6,Namibia Vs Sri Lanka,Namibia,7,David Wiese,0,1,0,0,0.00,out
7,Namibia Vs Sri Lanka,Namibia,8,JJ Smit,31,16,2,2,193.75,not_out
8,Namibia Vs Sri Lanka,Sri Lanka,1,Pathum Nissanka,9,10,1,0,90.00,out
9,Namibia Vs Sri Lanka,Sri Lanka,2,Kusal Mendisâ€,6,6,0,0,100.00,out


In [73]:
df_batting['match_id'] = df_batting['match'].map(match_ids_dict) 
df_batting.loc[-5:]

,match,teamInnings,battingPos,batsmanName,runs,balls,4s,6s,SR,out/not_out,match_id
0,Namibia Vs Sri Lanka,Namibia,1,Michael van Lingen,3,6,0,0,50.00,out,T20I # 1823
1,Namibia Vs Sri Lanka,Namibia,2,Divan la Cock,9,9,1,0,100.00,out,T20I # 1823
2,Namibia Vs Sri Lanka,Namibia,3,Jan Nicol Loftie-Eaton,20,12,1,2,166.66,out,T20I # 1823
3,Namibia Vs Sri Lanka,Namibia,4,Stephan Baard,26,24,2,0,108.33,out,T20I # 1823
4,Namibia Vs Sri Lanka,Namibia,5,Gerhard Erasmus(c),20,24,0,0,83.33,out,T20I # 1823
...,...,...,...,...,...,...,...,...,...,...,...
694,Pakistan Vs England,England,3,Phil Salt,10,9,2,0,111.11,out,T20I # 1879
695,Pakistan Vs England,England,4,Ben Stokes,52,49,5,1,106.12,not_out,T20I # 1879
696,Pakistan Vs England,England,5,Harry Brook,20,23,1,0,86.95,out,T20I # 1879
697,Pakistan Vs England,England,6,Moeen Ali,19,13,3,0,146.15,out,T20I # 1879


Cleanup strings by removing special characters from names

In [74]:
df_batting['batsmanName'] = df_batting['batsmanName'].apply(lambda x: x.replace('â€', ''))
df_batting['batsmanName'] = df_batting['batsmanName'].apply(lambda x: x.replace('\xa0', ''))
df_batting['batsmanName'] = df_batting['batsmanName'].apply(lambda x: x.replace('(c)', ''))

df_batting.head()

,match,teamInnings,battingPos,batsmanName,runs,balls,4s,6s,SR,out/not_out,match_id
0,Namibia Vs Sri Lanka,Namibia,1,Michael van Lingen,3,6,0,0,50.00,out,T20I # 1823
1,Namibia Vs Sri Lanka,Namibia,2,Divan la Cock,9,9,1,0,100.00,out,T20I # 1823
2,Namibia Vs Sri Lanka,Namibia,3,Jan Nicol Loftie-Eaton,20,12,1,2,166.66,out,T20I # 1823
3,Namibia Vs Sri Lanka,Namibia,4,Stephan Baard,26,24,2,0,108.33,out,T20I # 1823
4,Namibia Vs Sri Lanka,Namibia,5,Gerhard Erasmus,20,24,0,0,83.33,out,T20I # 1823


Renaming column "out/not_out" to "out"
out = 1
not_out = 0

In [75]:
df_batting["out"] = df_batting['out/not_out'].apply(lambda x: 1 if x=="out" else 0)
df_batting.head(11)

,match,teamInnings,battingPos,batsmanName,runs,balls,4s,6s,SR,out/not_out,match_id,out
0,Namibia Vs Sri Lanka,Namibia,1,Michael van Lingen,3,6,0,0,50.00,out,T20I # 1823,1
1,Namibia Vs Sri Lanka,Namibia,2,Divan la Cock,9,9,1,0,100.00,out,T20I # 1823,1
2,Namibia Vs Sri Lanka,Namibia,3,Jan Nicol Loftie-Eaton,20,12,1,2,166.66,out,T20I # 1823,1
3,Namibia Vs Sri Lanka,Namibia,4,Stephan Baard,26,24,2,0,108.33,out,T20I # 1823,1
4,Namibia Vs Sri Lanka,Namibia,5,Gerhard Erasmus,20,24,0,0,83.33,out,T20I # 1823,1
5,Namibia Vs Sri Lanka,Namibia,6,Jan Frylinck,44,28,4,0,157.14,out,T20I # 1823,1
6,Namibia Vs Sri Lanka,Namibia,7,David Wiese,0,1,0,0,0.00,out,T20I # 1823,1
7,Namibia Vs Sri Lanka,Namibia,8,JJ Smit,31,16,2,2,193.75,not_out,T20I # 1823,0
8,Namibia Vs Sri Lanka,Sri Lanka,1,Pathum Nissanka,9,10,1,0,90.00,out,T20I # 1823,1
9,Namibia Vs Sri Lanka,Sri Lanka,2,Kusal Mendis,6,6,0,0,100.00,out,T20I # 1823,1


In [76]:
df_batting.drop('out/not_out',axis=1,inplace=True)
df_batting.head()

,match,teamInnings,battingPos,batsmanName,runs,balls,4s,6s,SR,match_id,out
0,Namibia Vs Sri Lanka,Namibia,1,Michael van Lingen,3,6,0,0,50.00,T20I # 1823,1
1,Namibia Vs Sri Lanka,Namibia,2,Divan la Cock,9,9,1,0,100.00,T20I # 1823,1
2,Namibia Vs Sri Lanka,Namibia,3,Jan Nicol Loftie-Eaton,20,12,1,2,166.66,T20I # 1823,1
3,Namibia Vs Sri Lanka,Namibia,4,Stephan Baard,26,24,2,0,108.33,T20I # 1823,1
4,Namibia Vs Sri Lanka,Namibia,5,Gerhard Erasmus,20,24,0,0,83.33,T20I # 1823,1


Changing the column name '4s' to 'fours' and the column name '6s' to 'sixes'.

In [77]:
df_batting.rename(columns={'4s':'fours','6s':'sixes'},inplace=True)

In [78]:
df_batting.head()

,match,teamInnings,battingPos,batsmanName,runs,balls,fours,sixes,SR,match_id,out
0,Namibia Vs Sri Lanka,Namibia,1,Michael van Lingen,3,6,0,0,50.00,T20I # 1823,1
1,Namibia Vs Sri Lanka,Namibia,2,Divan la Cock,9,9,1,0,100.00,T20I # 1823,1
2,Namibia Vs Sri Lanka,Namibia,3,Jan Nicol Loftie-Eaton,20,12,1,2,166.66,T20I # 1823,1
3,Namibia Vs Sri Lanka,Namibia,4,Stephan Baard,26,24,2,0,108.33,T20I # 1823,1
4,Namibia Vs Sri Lanka,Namibia,5,Gerhard Erasmus,20,24,0,0,83.33,T20I # 1823,1


In [79]:
df_batting.dtypes

match          object
teamInnings    object
battingPos      int64
batsmanName    object
runs           object
balls          object
fours          object
sixes          object
SR             object
match_id       object
out             int64
dtype: object

In [80]:
df_batting.to_csv('cricket_csv_files/batting_summary.csv',index=False)

In [81]:
text = "Café, Über, Niño! Juhi†nuyb\xa0"

# Keep only alphabetic characters
clean_text = ''.join(char for char in text if char.isalpha() or char ==" ")

print(clean_text)


Café Über Niño Juhiâ nuyb


Bowling Summary

In [39]:
with open("E:\\BSc.IT\\TY\\DS Project\\t20_json_files\\t20_wc_bowling_summary.json") as f:
    data = json.load(f)
    all_rec = []
    for rec in data:
        all_rec.extend(rec["bowlingSummary"])
all_rec[:2]         


[{'match': 'Namibia Vs Sri Lanka',
  'bowlingTeam': 'Sri Lanka',
  'bowlerName': 'Maheesh Theekshana',
  'overs': '4',
  'maiden': '0',
  'runs': '23',
  'wickets': '1',
  'economy': '5.75',
  '0s': '7',
  '4s': '0',
  '6s': '0',
  'wides': '2',
  'noBalls': '0'},
 {'match': 'Namibia Vs Sri Lanka',
  'bowlingTeam': 'Sri Lanka',
  'bowlerName': 'Dushmantha Chameera',
  'overs': '4',
  'maiden': '0',
  'runs': '39',
  'wickets': '1',
  'economy': '9.75',
  '0s': '6',
  '4s': '3',
  '6s': '1',
  'wides': '2',
  'noBalls': '0'}]

In [40]:
df_bowling = pd.DataFrame(all_rec)
print(df_bowling.shape)

(500, 13)


In [41]:
df_bowling.head()

,match,bowlingTeam,bowlerName,overs,maiden,runs,wickets,economy,0s,4s,6s,wides,noBalls
0,Namibia Vs Sri Lanka,Sri Lanka,Maheesh Theekshana,4,0,23,1,5.75,7,0,0,2,0
1,Namibia Vs Sri Lanka,Sri Lanka,Dushmantha Chameera,4,0,39,1,9.75,6,3,1,2,0
2,Namibia Vs Sri Lanka,Sri Lanka,Pramod Madushan,4,0,37,2,9.25,6,3,1,0,0
3,Namibia Vs Sri Lanka,Sri Lanka,Chamika Karunaratne,4,0,36,1,9.00,7,3,1,1,0
4,Namibia Vs Sri Lanka,Sri Lanka,Wanindu Hasaranga de Silva,4,0,27,1,6.75,8,1,1,0,0


In [32]:
df_bowling["match_id"]=df_bowling["match"].map(match_ids_dict)
df_bowling.head()

,match,bowlingTeam,bowlerName,overs,maiden,runs,wickets,economy,0s,4s,6s,wides,noBalls,match_id
0,Namibia Vs Sri Lanka,Sri Lanka,Maheesh Theekshana,4,0,23,1,5.75,7,0,0,2,0,T20I # 1823
1,Namibia Vs Sri Lanka,Sri Lanka,Dushmantha Chameera,4,0,39,1,9.75,6,3,1,2,0,T20I # 1823
2,Namibia Vs Sri Lanka,Sri Lanka,Pramod Madushan,4,0,37,2,9.25,6,3,1,0,0,T20I # 1823
3,Namibia Vs Sri Lanka,Sri Lanka,Chamika Karunaratne,4,0,36,1,9.00,7,3,1,1,0,T20I # 1823
4,Namibia Vs Sri Lanka,Sri Lanka,Wanindu Hasaranga de Silva,4,0,27,1,6.75,8,1,1,0,0,T20I # 1823


In [33]:
df_bowling.dtypes

match          object
bowlingTeam    object
bowlerName     object
overs          object
maiden         object
runs           object
wickets        object
economy        object
0s             object
4s             object
6s             object
wides          object
noBalls        object
match_id       object
dtype: object

In [41]:
overs = '2.4'

bals = overs.split('.')[1]
bals

#overs = 2.5
#after_dec = (overs-round(overs))*10
#int(after_dec)
#round(overs)*6 

'4'

In [42]:
df_bowling['overs']=df_bowling['overs'].str.split('.')
df_bowling.head(32)

,match,bowlingTeam,bowlerName,overs,maiden,runs,wickets,economy,0s,4s,6s,wides,noBalls
0,Namibia Vs Sri Lanka,Sri Lanka,Maheesh Theekshana,[4],0,23,1,5.75,7,0,0,2,0
1,Namibia Vs Sri Lanka,Sri Lanka,Dushmantha Chameera,[4],0,39,1,9.75,6,3,1,2,0
2,Namibia Vs Sri Lanka,Sri Lanka,Pramod Madushan,[4],0,37,2,9.25,6,3,1,0,0
3,Namibia Vs Sri Lanka,Sri Lanka,Chamika Karunaratne,[4],0,36,1,9.00,7,3,1,1,0
4,Namibia Vs Sri Lanka,Sri Lanka,Wanindu Hasaranga de Silva,[4],0,27,1,6.75,8,1,1,0,0
5,Namibia Vs Sri Lanka,Namibia,Gerhard Erasmus,[1],0,8,0,8.00,1,1,0,0,0
6,Namibia Vs Sri Lanka,Namibia,David Wiese,[4],0,16,2,4.00,13,1,0,0,0
7,Namibia Vs Sri Lanka,Namibia,Bernard Scholtz,[4],0,18,2,4.50,10,1,0,0,0
8,Namibia Vs Sri Lanka,Namibia,Ben Shikongo,[3],1,22,2,7.33,6,3,0,0,0
9,Namibia Vs Sri Lanka,Namibia,JJ Smit,[3],0,16,1,5.33,7,0,0,1,0


In [43]:
df_bowling['overs1']=df_bowling['overs'].apply(lambda x: x[0])
df_bowling.head(25)

,match,bowlingTeam,bowlerName,overs,maiden,runs,wickets,economy,0s,4s,6s,wides,noBalls,overs1
0,Namibia Vs Sri Lanka,Sri Lanka,Maheesh Theekshana,[4],0,23,1,5.75,7,0,0,2,0,4
1,Namibia Vs Sri Lanka,Sri Lanka,Dushmantha Chameera,[4],0,39,1,9.75,6,3,1,2,0,4
2,Namibia Vs Sri Lanka,Sri Lanka,Pramod Madushan,[4],0,37,2,9.25,6,3,1,0,0,4
3,Namibia Vs Sri Lanka,Sri Lanka,Chamika Karunaratne,[4],0,36,1,9.00,7,3,1,1,0,4
4,Namibia Vs Sri Lanka,Sri Lanka,Wanindu Hasaranga de Silva,[4],0,27,1,6.75,8,1,1,0,0,4
5,Namibia Vs Sri Lanka,Namibia,Gerhard Erasmus,[1],0,8,0,8.00,1,1,0,0,0,1
6,Namibia Vs Sri Lanka,Namibia,David Wiese,[4],0,16,2,4.00,13,1,0,0,0,4
7,Namibia Vs Sri Lanka,Namibia,Bernard Scholtz,[4],0,18,2,4.50,10,1,0,0,0,4
8,Namibia Vs Sri Lanka,Namibia,Ben Shikongo,[3],1,22,2,7.33,6,3,0,0,0,3
9,Namibia Vs Sri Lanka,Namibia,JJ Smit,[3],0,16,1,5.33,7,0,0,1,0,3


In [44]:
df_bowling['overs2']=df_bowling['overs'].apply(lambda x: x[1] if len(x)==2 else 0)
df_bowling.head(25)

,match,bowlingTeam,bowlerName,overs,maiden,runs,wickets,economy,0s,4s,6s,wides,noBalls,overs1,overs2
0,Namibia Vs Sri Lanka,Sri Lanka,Maheesh Theekshana,[4],0,23,1,5.75,7,0,0,2,0,4,0
1,Namibia Vs Sri Lanka,Sri Lanka,Dushmantha Chameera,[4],0,39,1,9.75,6,3,1,2,0,4,0
2,Namibia Vs Sri Lanka,Sri Lanka,Pramod Madushan,[4],0,37,2,9.25,6,3,1,0,0,4,0
3,Namibia Vs Sri Lanka,Sri Lanka,Chamika Karunaratne,[4],0,36,1,9.00,7,3,1,1,0,4,0
4,Namibia Vs Sri Lanka,Sri Lanka,Wanindu Hasaranga de Silva,[4],0,27,1,6.75,8,1,1,0,0,4,0
5,Namibia Vs Sri Lanka,Namibia,Gerhard Erasmus,[1],0,8,0,8.00,1,1,0,0,0,1,0
6,Namibia Vs Sri Lanka,Namibia,David Wiese,[4],0,16,2,4.00,13,1,0,0,0,4,0
7,Namibia Vs Sri Lanka,Namibia,Bernard Scholtz,[4],0,18,2,4.50,10,1,0,0,0,4,0
8,Namibia Vs Sri Lanka,Namibia,Ben Shikongo,[3],1,22,2,7.33,6,3,0,0,0,3,0
9,Namibia Vs Sri Lanka,Namibia,JJ Smit,[3],0,16,1,5.33,7,0,0,1,0,3,0


In [92]:
df_bowling.shape

(500, 16)

In [37]:
df_bowling.rename(columns={'0s':'zeros','4s':'fours','6s':'sixes'},inplace=True)
df_bowling.head(3)

,match,bowlingTeam,bowlerName,overs,maiden,runs,wickets,economy,zeros,fours,sixes,wides,noBalls,match_id,overs1,overs2
0,Namibia Vs Sri Lanka,Sri Lanka,Maheesh Theekshana,[4],0,23,1,5.75,7,0,0,2,0,T20I # 1823,4,0
1,Namibia Vs Sri Lanka,Sri Lanka,Dushmantha Chameera,[4],0,39,1,9.75,6,3,1,2,0,T20I # 1823,4,0
2,Namibia Vs Sri Lanka,Sri Lanka,Pramod Madushan,[4],0,37,2,9.25,6,3,1,0,0,T20I # 1823,4,0


In [45]:
df_bowling.to_csv('cricket_csv_files/bowling_summary.csv',index=False)

Players Summary


In [50]:
with open("E:\\BSc.IT\\TY\\DS Project\\t20_json_files\\t20_wc_player_info.json") as f:
    data = json.load(f)


In [51]:
df_players = pd.DataFrame(data)
print(df_players.shape)
df_players.head()

(219, 6)


,name,team,battingStyle,bowlingStyle,playingRole,description
0,Michael van Lingen,Namibia,Left hand Bat,Left arm Medium,Bowling Allrounder,
1,Divan la Cock,Namibia,Right hand Bat,Legbreak,Opening Batter,
2,Jan Nicol Loftie-Eaton,Namibia,Left hand Bat,"Right arm Medium, Legbreak",Batter,
3,Stephan Baard,Namibia,Right hand Bat,Right arm Medium fast,Batter,
4,Gerhard Erasmus(c),Namibia,Right hand Bat,Right arm Offbreak,Allrounder,


In [52]:
df_players['name'] = df_players['name'].apply(lambda x: x.replace('â€', ''))
df_players['name'] = df_players['name'].apply(lambda x: x.replace('†', ''))
df_players['name'] = df_players['name'].apply(lambda x: x.replace('\xa0', ''))
df_players['name'] = df_players['name'].apply(lambda x: x.replace('(c)', ''))
df_players.head(10)

,name,team,battingStyle,bowlingStyle,playingRole,description
0,Michael van Lingen,Namibia,Left hand Bat,Left arm Medium,Bowling Allrounder,
1,Divan la Cock,Namibia,Right hand Bat,Legbreak,Opening Batter,
2,Jan Nicol Loftie-Eaton,Namibia,Left hand Bat,"Right arm Medium, Legbreak",Batter,
3,Stephan Baard,Namibia,Right hand Bat,Right arm Medium fast,Batter,
4,Gerhard Erasmus,Namibia,Right hand Bat,Right arm Offbreak,Allrounder,
5,Jan Frylinck,Namibia,Left hand Bat,Left arm Fast medium,Allrounder,
6,David Wiese,Namibia,Right hand Bat,Right arm Medium fast,Allrounder,David Wiese joined a marked outflow of South A...
7,JJ Smit,Namibia,Right hand Bat,Left arm Medium fast,Bowling Allrounder,
8,Pathum Nissanka,Sri Lanka,Right hand Bat,,Top order Batter,
9,Kusal Mendis,Sri Lanka,Right hand Bat,Legbreak,Wicketkeeper Batter,"Blessed with a compact technique, an aggressiv..."


In [53]:
df_players[df_players['team'] == 'India']


,name,team,battingStyle,bowlingStyle,playingRole,description
127,KL Rahul,India,Right hand Bat,,Opening Batter,"A tall, elegant right-hand batsman who can kee..."
128,Rohit Sharma,India,Right hand Bat,Right arm Offbreak,Top order Batter,"Languid and easy on the eye, Rohit Sharma owne..."
129,Virat Kohli,India,Right hand Bat,Right arm Medium,Top order Batter,India has given to the world many a great cric...
130,Suryakumar Yadav,India,Right hand Bat,"Right arm Medium, Right arm Offbreak",Batter,Hard-hitting 360-degree batter Suryakumar Yada...
131,Axar Patel,India,Left hand Bat,Slow Left arm Orthodox,Bowling Allrounder,Left-arm spinner Axar Patel has been increasin...
132,Hardik Pandya,India,Right hand Bat,Right arm Medium fast,Allrounder,Hardik Pandya swears by living life king size ...
133,Dinesh Karthik,India,Right hand Bat,Right arm Offbreak,Wicketkeeper Batter,Not many would forget the sight of Dinesh Kart...
134,Ravichandran Ashwin,India,Right hand Bat,Right arm Offbreak,Bowling Allrounder,R Ashwin took the tricks and skills he learned...
135,Bhuvneshwar Kumar,India,Right hand Bat,Right arm Medium,Bowler,"At the time of his India debut in 2012, Bhuvne..."
136,Arshdeep Singh,India,Left hand Bat,Left arm Medium fast,Bowler,


In [54]:
#df_players.to_csv('t20_csv_files/dim_players_no_images.csv', index = False)

create a search object containing a list of duplicated rows by calling pd.Dataframe.duplicated()

The duplicated row contains a *True* next to their row number.

The easiest wayy to determine which rows are duplicated is to create an index in which you use search == True
as the expression.

In [55]:
search = pd.DataFrame.duplicated(df_players)
print(df_players)
print()
print(search[search == True])

                       name          team    battingStyle  \
0        Michael van Lingen       Namibia   Left hand Bat   
1             Divan la Cock       Namibia  Right hand Bat   
2    Jan Nicol Loftie-Eaton       Namibia   Left hand Bat   
3             Stephan Baard       Namibia  Right hand Bat   
4           Gerhard Erasmus       Namibia  Right hand Bat   
..                      ...           ...             ...   
214          Tabraiz Shamsi  South Africa  Right hand Bat   
215               Phil Salt       England  Right hand Bat   
216            Chris Jordan       England  Right hand Bat   
217          Regis Chakabva      Zimbabwe  Right hand Bat   
218      Mehidy Hasan Miraz    Bangladesh  Right hand Bat   

                   bowlingStyle          playingRole  \
0               Left arm Medium   Bowling Allrounder   
1                      Legbreak       Opening Batter   
2    Right arm Medium, Legbreak               Batter   
3         Right arm Medium fast            

In [56]:
df_players.iloc[19]

name               Gerhard Erasmus
team                       Namibia
battingStyle        Right hand Bat
bowlingStyle    Right arm Offbreak
playingRole             Allrounder
description                       
Name: 19, dtype: object

In [57]:
#df_players.loc[df_players['name']=='Shakib Al Hasan']
df_players.loc[df_players['name']=='Gerhard Erasmus']

,name,team,battingStyle,bowlingStyle,playingRole,description
4,Gerhard Erasmus,Namibia,Right hand Bat,Right arm Offbreak,Allrounder,
19,Gerhard Erasmus,Namibia,Right hand Bat,Right arm Offbreak,Allrounder,


In [58]:
df_players.loc[df_players['name']=='Regis Chakabva']

,name,team,battingStyle,bowlingStyle,playingRole,description
35,Regis Chakabva,Zimbabwe,Right hand Bat,Right arm Offbreak,Wicketkeeper Batter,"A doughty wicketkeeper-batsman, Regis Chakabva..."
217,Regis Chakabva,Zimbabwe,Right hand Bat,Right arm Offbreak,Wicketkeeper Batter,"A doughty wicketkeeper-batsman, Regis Chakabva..."


In [59]:
df_players.loc[df_players['name']=='Matthew Wade']

,name,team,battingStyle,bowlingStyle,playingRole,description
179,Matthew Wade,Australia,Left hand Bat,Right arm Medium,Wicketkeeper Batter,It has been a career of reinvention for Matthe...
206,Matthew Wade,Australia,Left hand Bat,Right arm Medium,Wicketkeeper Batter,It has been a career of reinvention for Matthe...


In [60]:
df_players.shape

(219, 6)

# Removing duplicates

There are total 219 records with 6 duplicated records.

In [61]:
#print(df_players.drop_duplicates(inplace=True))
print(df_players.drop_duplicates())

                       name          team    battingStyle  \
0        Michael van Lingen       Namibia   Left hand Bat   
1             Divan la Cock       Namibia  Right hand Bat   
2    Jan Nicol Loftie-Eaton       Namibia   Left hand Bat   
3             Stephan Baard       Namibia  Right hand Bat   
4           Gerhard Erasmus       Namibia  Right hand Bat   
..                      ...           ...             ...   
213    Wellington Masakadza      Zimbabwe   Left hand Bat   
214          Tabraiz Shamsi  South Africa  Right hand Bat   
215               Phil Salt       England  Right hand Bat   
216            Chris Jordan       England  Right hand Bat   
218      Mehidy Hasan Miraz    Bangladesh  Right hand Bat   

                   bowlingStyle          playingRole  \
0               Left arm Medium   Bowling Allrounder   
1                      Legbreak       Opening Batter   
2    Right arm Medium, Legbreak               Batter   
3         Right arm Medium fast            

In [62]:
df_players.shape

(219, 6)

In [31]:
# df_batting.runs.dtype

dtype('O')

In [32]:
# df_batting["runs"]=df_batting["runs"].astype(int)
# df_batting.head(3)

,match,teamInnings,battingPos,batsmanName,runs,balls,4s,6s,SR,out/not_out,match_id
0,Namibia Vs Sri Lanka,Namibia,1,Michael van Lingen,3,6,0,0,50.00,out,T20I # 1823
1,Namibia Vs Sri Lanka,Namibia,2,Divan la Cock,9,9,1,0,100.00,out,T20I # 1823
2,Namibia Vs Sri Lanka,Namibia,3,Jan Nicol Loftie-Eaton,20,12,1,2,166.66,out,T20I # 1823


In [33]:
# df_batting.dtypes

match          object
teamInnings    object
battingPos      int64
batsmanName    object
runs            int64
balls          object
4s             object
6s             object
SR             object
out/not_out    object
match_id       object
dtype: object

In [34]:
# Openers = df_batting[df_batting["runs"]>30]

In [35]:
# Openers

,match,teamInnings,battingPos,batsmanName,runs,balls,4s,6s,SR,out/not_out,match_id
5,Namibia Vs Sri Lanka,Namibia,6,Jan Frylinck,44,28,4,0,157.14,out,T20I # 1823
7,Namibia Vs Sri Lanka,Namibia,8,JJ Smit,31,16,2,2,193.75,not_out,T20I # 1823
20,U.A.E. Vs Netherlands,U.A.E.,2,Muhammad Waseem,41,47,1,2,87.23,out,T20I # 1825
38,Scotland Vs West Indies,Scotland,1,George Munsey,66,53,9,0,124.52,not_out,T20I # 1826
51,Scotland Vs West Indies,West Indies,7,Jason Holder,38,33,4,1,115.15,out,T20I # 1826
...,...,...,...,...,...,...,...,...,...,...,...
680,India Vs England,England,1,Jos Buttler(c),80,49,9,3,163.26,not_out,T20I # 1878
681,India Vs England,England,2,Alex Hales,86,47,4,7,182.97,not_out,T20I # 1878
683,Pakistan Vs England,Pakistan,2,Babar Azam(c),32,28,2,0,114.28,out,T20I # 1879
685,Pakistan Vs England,Pakistan,4,Shan Masood,38,28,2,1,135.71,out,T20I # 1879


In [36]:
# from flask import Flask
# app = Flask(__name__)

# @app.route('/hello')
# def hello_world():
#    return 'Hello World in Flask'

# if __name__ == '__main__':
#    app.run()

 * Serving Flask app '__main__'
 * Debug mode: off


 * Running on http://127.0.0.1:5000
Press CTRL+C to quit
